scene/resources/style_box.cpp

```diff
@@ -709,23 +746,9 @@ void StyleBoxFlat::draw(RID p_canvas_item, const Rect2 &p_rect) const {
 
 	//DRAW INFILL
 	if (draw_center) {
-		int temp_vert_offset = verts.size();
 		int no_border[4] = { 0, 0, 0, 0 };
 		draw_ring(verts, indices, colors, style_rect, adapted_corner,
-				infill_rect, no_border, &bg_color, &bg_color, corner_detail);
-		int added_vert_count = verts.size() - temp_vert_offset;
-		//fill the indices and the colors for the center
-		for (int index = 0; index <= added_vert_count / 2; index += 2) {
-			int i = index;
-			//poly 1
-			indices.push_back(temp_vert_offset + i);
-			indices.push_back(temp_vert_offset + added_vert_count - 4 - i);
-			indices.push_back(temp_vert_offset + i + 2);
-			//poly 1
-			indices.push_back(temp_vert_offset + i);
-			indices.push_back(temp_vert_offset + added_vert_count - 2 - i);
-			indices.push_back(temp_vert_offset + added_vert_count - 4 - i);
-		}
+				infill_rect, no_border, &bg_color, &bg_color, corner_detail, true);
 	}
 
 	if (aa_on) {
```

Comment:
this is the actual issue of the one pixel light border. 
`Color c = border_color.read().ptr()[I];` The color is based on the border color. which is one correct if `border_width > 0`